src/Akka.Streams.Kafka/Settings/CommitterSettings.cs

```diff
@@ -1,25 +1,65 @@
 using System;
 using Akka.Actor;
 using Akka.Configuration;
+using Akka.Streams.Kafka.Messages;
 
 namespace Akka.Streams.Kafka.Settings
 {
+    /// <summary>
+    /// Determines when to commit offsets
+    /// </summary>
+    public interface ICommitWhen;
+
+    public static class CommitWhen
+    {
+        /// <summary>
+        /// Commit as soon as a <see cref="ICommittableOffset"/> is observed.
+        /// </summary>
+        public sealed class OffsetFirstObserved : ICommitWhen
+        {
+            public static OffsetFirstObserved Instance { get; } = new OffsetFirstObserved();
+            private OffsetFirstObserved() { }
+        }
+        
+        /// <summary>
+        /// Commit the previous offset as soon as a new <see cref="ICommittableOffset"/> is observed.
+        /// </summary>
+        public sealed class NextOffsetObserved : ICommitWhen
+        {
+            public static NextOffsetObserved Instance { get; } = new NextOffsetObserved();
+            private NextOffsetObserved() { }
+        }
+
+        public static ICommitWhen ValueOf(string s)
+        {
+            return s.ToLowerInvariant() switch
+            {
+                "offset-first-observed" => OffsetFirstObserved.Instance,
+                "next-offset-observed" => NextOffsetObserved.Instance,
+                _ => throw new ArgumentException(
+                    $"Allow values are offset-first-observed, next-offset-observed.Received: {s}")
+            };
+        }
+    }
+    
     /// <summary>
     /// Settings for committer. See 'akka.kafka.committer' section in reference.conf.
     /// </summary>
-    public sealed class CommitterSettings
+    public sealed record CommitterSettings
```

Comment:
Turning stuff into `record`s as part of the public API to cut down on the amount of copying / equality stuff going on.